scripts/dataset_processing/tts/audio_processing/config/preprocessing.yaml

```diff
@@ -0,0 +1,19 @@
+name: "preprocessing"
+
+data_base_dir: "/home"
+
+defaults:
+  - trim: energy
+
+config:
+  _target_: scripts.dataset_processing.tts.audio_processing.preprocess_audio.AudioPreprocessingConfig
+  input_manifest: ${data_base_dir}/manifest.json
+  output_manifest: ${data_base_dir}/manifest_processed.json
+  output_dir: ${data_base_dir}/audio_processed
+  num_workers: -1
+  max_entries: 0
+  output_sample_rate: 0
+  volume_level: 0.95
+  min_duration: 0.5
+  max_duration: 10.0
```

Comment:
In the end these are just examples that people will want to tweak depending on their dataset.

An upper bound of 10 seconds is a rule of thumb. I don't remember where I heard it originally, but I heard the same thing from Boris/Evelina months ago when we were talking about creating new HiFi-TTS like datasets. Its mainly done for computation reasons; the compute time and memory for a training batch is primarily dependent on the longest sequence in the batch. To use utterances much longer than 10 seconds you typically implement duration based batching (eg. there are a lot of papers that claim they use batch sizes of 10k or 20k audio frames, instead of fixed utterance count). A reason it comes up often is because TTS datasets are often derived from ASR datasets, and ASR likes training with utterances up to 20 seconds long.

The lower bound is more arbitrary. I would be OK with setting it to 0.1 for consistency. The reason I would set it higher than 0.1 here is to help identify/remove edge cases where the trimming ends up removing too much of the utterance. Alternatively I considered storing the original and trimmed durations in the output manifest and leaving the duration filtering/validation/analysis as a second step/script (the filtering could be done with a simple JQ command https://stedolan.github.io/jq/), but thought that was too complicated for an initial implementation.

Reply:
In general I'd be worried that people will blindly take whatever parameters are set in the example as defaults :P The reasoning is good and these seem fine, though.

Reply:
thanks for explanation. how about make it consistent with other yamls, and leave a comment to user?